### ISSUE TITLE
Mismatch issue in SAD loop unit test

### ISSUE THREAD

**Issue by Cidana-Developers** [OpenVisualCloud/SVT-AV1#771]
Step to reproduce:
~~~
./SvtAv1UnitTests --gtest_filter="LOOPSAD*"
~~~

Console output (part of failed cases and final results):
~~~
[ RUN      ] LOOPSAD/SadLoopTest.SadLoopTest/12410
C:\Works\AV1\SVT-AV1\test\SadTest.cc(729): error: Expected equality of these values:
  best_sad0
    Which is: 171981
  best_sad1
    Which is: 171093
compare best_sad error block dim: [64 x 32] search area [15 x 6]
C:\Works\AV1\SVT-AV1\test\SadTest.cc(734): error: Expected equality of these values:
  x_search_center0
    Which is: 0
  x_search_center1
    Which is: 20
compare x_search_center error block dim: [64 x 32] search area [15 x 6]
[  FAILED  ] LOOPSAD/SadLoopTest.SadLoopTest/12410, where GetParam() = (3, (64, 32), (15, 6), (00007FF7E10D31B0, 00007FF7E110FD80)) (48 ms)
[ RUN      ] LOOPSAD/SadLoopTest.SadLoopTest/12411
C:\Works\AV1\SVT-AV1\test\SadTest.cc(729): error: Expected equality of these values:
  best_sad0
    Which is: 171981
  best_sad1
    Which is: 171093
compare best_sad error block dim: [64 x 32] search area [15 x 6]
C:\Works\AV1\SVT-AV1\test\SadTest.cc(734): error: Expected equality of these values:
  x_search_center0
    Which is: 0
  x_search_center1
    Which is: 20
compare x_search_center error block dim: [64 x 32] search area [15 x 6]
[  FAILED  ] LOOPSAD/SadLoopTest.SadLoopTest/12411, where GetParam() = (3, (64, 32), (15, 6), (00007FF7E10D31B0, 00007FF7E11482A0)) (49 ms)

[==========] 16240 tests from 1 test case ran. (739354 ms total)
[  PASSED  ] 16222 tests.
[  FAILED  ] 18 tests, listed below:
[  FAILED  ] LOOPSAD/SadLoopTest.SadLoopTest/8234, where GetParam() = (2, (64, 64), (15, 6), (00007FF7E10D31B0, 00007FF7E110FD80))
[  FAILED  ] LOOPSAD/SadLoopTest.SadLoopTest/8235, where GetParam() = (2, (64, 64), (15, 6), (00007FF7E10D31B0, 00007FF7E11482A0))
[  FAILED  ] LOOPSAD/SadLoopTest.SadLoopTest/8698, where GetParam() = (2, (32, 16), (15, 6), (00007FF7E10D31B0, 00007FF7E110FD80))
[  FAILED  ] LOOPSAD/SadLoopTest.SadLoopTest/8699, where GetParam() = (2, (32, 16), (15, 6), (00007FF7E10D31B0, 00007FF7E11482A0))
[  FAILED  ] LOOPSAD/SadLoopTest.SadLoopTest/9278, where GetParam() = (2, (8, 8), (15, 6), (00007FF7E10D31B0, 00007FF7E110FD80))
[  FAILED  ] LOOPSAD/SadLoopTest.SadLoopTest/9279, where GetParam() = (2, (8, 8), (15, 6), (00007FF7E10D31B0, 00007FF7E11482A0))
[  FAILED  ] LOOPSAD/SadLoopTest.SadLoopTest/9858, where GetParam() = (2, (16, 4), (15, 6), (00007FF7E10D31B0, 00007FF7E110FD80))
[  FAILED  ] LOOPSAD/SadLoopTest.SadLoopTest/9859, where GetParam() = (2, (16, 4), (15, 6), (00007FF7E10D31B0, 00007FF7E11482A0))
[  FAILED  ] LOOPSAD/SadLoopTest.SadLoopTest/10322, where GetParam() = (2, (64, 16), (15, 6), (00007FF7E10D31B0, 00007FF7E110FD80))
[  FAILED  ] LOOPSAD/SadLoopTest.SadLoopTest/10323, where GetParam() = (2, (64, 16), (15, 6), (00007FF7E10D31B0, 00007FF7E11482A0))
[  FAILED  ] LOOPSAD/SadLoopTest.SadLoopTest/11134, where GetParam() = (2, (48, 24), (15, 6), (00007FF7E10D31B0, 00007FF7E110FD80))
[  FAILED  ] LOOPSAD/SadLoopTest.SadLoopTest/11135, where GetParam() = (2, (48, 24), (15, 6), (00007FF7E10D31B0, 00007FF7E11482A0))
[  FAILED  ] LOOPSAD/SadLoopTest.SadLoopTest/12294, where GetParam() = (3, (64, 64), (15, 6), (00007FF7E10D31B0, 00007FF7E110FD80))
[  FAILED  ] LOOPSAD/SadLoopTest.SadLoopTest/12295, where GetParam() = (3, (64, 64), (15, 6), (00007FF7E10D31B0, 00007FF7E11482A0))
[  FAILED  ] LOOPSAD/SadLoopTest.SadLoopTest/12410, where GetParam() = (3, (64, 32), (15, 6), (00007FF7E10D31B0, 00007FF7E110FD80))
[  FAILED  ] LOOPSAD/SadLoopTest.SadLoopTest/12411, where GetParam() = (3, (64, 32), (15, 6), (00007FF7E10D31B0, 00007FF7E11482A0))
[  FAILED  ] LOOPSAD/SadLoopTest.SadLoopTest/15078, where GetParam() = (3, (64, 24), (15, 6), (00007FF7E10D31B0, 00007FF7E110FD80))
[  FAILED  ] LOOPSAD/SadLoopTest.SadLoopTest/15079, where GetParam() = (3, (64, 24), (15, 6), (00007FF7E10D31B0, 00007FF7E11482A0))

18 FAILED TESTS
~~~

target functions:
~~~
 - sad_loop_kernel_sparse_sse4_1_intrin
 - sad_loop_kernel_sparse_avx2_intrin
~~~

**Comment by ttrigui** [comment]
@spawlows Can you please look into this issue?

**Comment by spawlows** [comment]
Fixed PR #806
